raster_analysis/window.py

```diff
@@ -93,11 +93,6 @@ def _read_window(
                         data = vrt.read(1, masked=masked, window=window)
                         no_data_value = src.nodata
 
-                        # convert nan to 0 if no data value is set no nan
```

Comment:
@solomon-negusse @gtempus @dmannarino it turns out this is the reason the tests didn't catch the NaN issue - there was existing code that converts NaN values to 0 on reading the raster, so NaN values would never make it to the later checks. 

Once I removed this, the tests with floats started failing, and now I fixed it by adding some additional checks it'd be good for you to rereview.

Reply:
Great, thanks for hunting this down!